File: src/GameLogger.Core/Data/Game.cs
namespace GameLogger.Core.Data;

public sealed class Game
{
    public Guid Id { get; init; }
    public string Title { get; init; }
    public DateOnly StartDate { get; init; }
    public DateOnly? FinishDate { get; init; }
    public string? Platform { get; init; }
    public ushort? HoursPlayed { get; init; }
    public ushort? Rating { get; init; }
    public string? Notes { get; init; }

    private bool Equals(Game other)
    {
        return Id.Equals(other.Id);
    }

    public override bool Equals(object? obj)
    {
        return ReferenceEquals(this, obj) || obj is Game other && Equals(other);
    }

    public override int GetHashCode()
    {
        return Id.GetHashCode();
    }
}

File: pkg/model/game.go
package model

import "time"

type Game struct {
	Id           string        `json:"id"`
	Title        string        `json:"title"`
	Playthroughs []Playthrough `json:"playthroughs"`
}

type Playthrough struct {
	StartDate   time.Time  `json:"start_date"`
	FinishDate  *time.Time `json:"finish_date,omitempty"`
	Platform    string     `json:"platform,omitempty"`
	HoursPlayed *int       `json:"hours_played,omitempty"`
	Rating      *int       `json:"rating,omitempty"`
	Notes       string     `json:"notes,omitempty"`
}

func (g *Game) AddPlaythrough(pt Playthrough) {
	g.Playthroughs = append(g.Playthroughs, pt)
}
